### PR TITLE
Allow for transformed PWMOut-based error

### LINKED ISSUE #66: Stepper motor assumes PWM pins are variable frequency
When initializing a stepper with PWM pins, the stepper attempts to raise the PWM frequency to 2000 Hz if it is set to below 1500 Hz. This happens frequently as PWMOut pins default to 500 Hz. However, the stepper doesn't check that these values are writable (that the PWM pin's `vairable_frequency` is `True`) before writing, resulting in a very vague AttributeError when initialized with a default PWMOut. This should be pretty easy thing to check and throw a more descriptive error. The error happens on line 117 in stepper.py.

### PR DESCRIPTION
**Pull request by tekktrik**
Fixes #66

### CODE REVIEW

File: adafruit_motor/stepper.py [hunk 1]
```diff
@@ -113,8 +113,13 @@ def __init__(
             # set a safe pwm freq for each output
             self._coil = (ain2, bin1, ain1, bin2)
             for i in range(4):
-                if self._coil[i].frequency < 1500:
+                if self._coil[i].frequency < 1500 and self._coil[i].variable_frequency:
                     self._coil[i].frequency = 2000
+                else:
+                    raise RuntimeError(
```
Comment (dhalbert) [inline adafruit_motor/stepper.py]: This is more of a "bad arguments" error than some uncontrollable failure at run-time, so I suggest using `ValueError`.

```suggestion
                    raise ValueError(
```

**Review by dhalbert — DISMISSED** [review]
Good fix! One suggestion.

File: adafruit_motor/stepper.py (hunk 1 above)
Comment (tekktrik) [reply adafruit_motor/stepper.py]: Good idea, that's a better fit.

File: adafruit_motor/stepper.py [hunk 2]
```diff
@@ -113,8 +113,13 @@ def __init__(
             # set a safe pwm freq for each output
             self._coil = (ain2, bin1, ain1, bin2)
             for i in range(4):
-                if self._coil[i].frequency < 1500:
+                if self._coil[i].frequency < 1500 and self._coil[i].variable_frequency:
                     self._coil[i].frequency = 2000
+                else:
```
Comment (jepler) [inline adafruit_motor/stepper.py]: Won't this error fire if the frequency is above 1500 but not variable? Or do I misread it?

Comment (tekktrik) [reply adafruit_motor/stepper.py]: You are absolutely correct, the logic should be inverted to use a guard clause instead for `if under and not variable: raise error`.

**Review by jepler — APPROVED** [review]
That looks sensible now, thank you!

**Pull request merged**